src/pudl/transform/ferc714.py

```diff
@@ -0,0 +1,749 @@
+    new_df["local_datetime"] = (
+        new_df
+        .groupby("timezone")["utc_aware"]
+        .transform(lambda x: x.dt.tz_convert(x.name))
+    )
+
+    # We can only use the Series.dt datetime accessor on a Series with
+    # homogeneous timezone information, so we now have to iterate through each
+    # of the timezones. If there's a better way to do this with groupby that
+    # would be great...
+    for tz in new_df.timezone.unique():
+        rows_to_fix = (new_df.timezone == tz) & (new_df.report_year.isnull())
+        if rows_to_fix.any():
+            new_df.loc[rows_to_fix, "report_year"] = \
+                new_df.loc[rows_to_fix, "local_datetime"].dt.year
+    # Remove temporary columns and return only the columns we started with.
+    new_df = new_df.drop(["utc_aware", "local_datetime"], axis="columns")
+    non_null = len(new_df[new_df.demand_mwh.notnull()]) / len(new_df)
+
+    logger.info(
+        f"{non_null:.2%} of all planning area demand records have "
+        "non-null values.")
+
+    return new_df
+
+
+def respondent_id(tf_dfs):
+    """
+    Transform the FERC 714 respondent IDs, names, and EIA utility IDs.
+
+    This consists primarily of dropping test respondents and manually
+    assigning EIA utility IDs to a few FERC Form 714 respondents that report
+    planning area demand, but which don't have their corresponding EIA utility
+    IDs provided by FERC for some reason (including PacifiCorp).
+
+    Args:
+        tf_dfs (dict): A dictionary of (partially) transformed dataframes,
+            to be cleaned up.
+
+    Returns:
+        dict: The input dictionary of dataframes, but with a finished
+        respondent_id_ferc714 dataframe.
+
+    """
+    df = (
+        tf_dfs["respondent_id_ferc714"].assign(
+            utility_name_ferc714=lambda x: x.utility_name_ferc714.str.strip(),
+            utility_id_eia=lambda x: x.utility_id_eia.replace(
+                to_replace=0, value=pd.NA)
+        )
+        # These excludes fake Test IDs -- not real planning areas
+        .query("utility_id_ferc714 not in @BAD_RESPONDENTS")
+    )
+    # There are a few utilities that seem mappable, but missing:
+    for rid in MISSING_UTILITY_ID_EIA:
+        df.loc[df.utility_id_ferc714 == rid,
+               "utility_id_eia"] = MISSING_UTILITY_ID_EIA[rid]
+    tf_dfs["respondent_id_ferc714"] = df
+    return tf_dfs
+
+
+def pa_demand_hourly(tf_dfs):
+    """
+    Transform the hourly demand time series by Planning Area.
+
+    Args:
+        tf_dfs (dict): A dictionary of (partially) transformed dataframes,
+            to be cleaned up.
+
+    Returns:
+        dict: The input dictionary of dataframes, but with a finished
+        pa_demand_hourly_ferc714 dataframe.
+
+    """
+    logger.info("Converting dates into pandas Datetime types.")
+    df = (
+        tf_dfs["pa_demand_hourly_ferc714"].assign(
+            report_date=lambda x: pd.to_datetime(x.report_date),
```

Comment:
for `report_date` why are you doing this in an `assign` versus just using `astype`? No real preference, but astype seems more straightforward.